packages/babel-parser/src/parser/statement.js

```diff
@@ -838,14 +847,20 @@ export default class StatementParser extends ExpressionParser {
     directives: ?(N.Directive[]),
     topLevel: boolean,
     end: TokenType,
+    afterBlockParse?: (hasStrictModeDirective: boolean) => void,
   ): void {
+    const octalPositions = [];
     let parsedNonDirective = false;
-    let oldStrict;
-    let octalPosition;
+    let oldStrict = null;
 
     while (!this.eat(end)) {
-      if (!parsedNonDirective && this.state.containsOctal && !octalPosition) {
-        octalPosition = this.state.octalPosition;
+      // Track octal literals that occur before a "use strict" directive.
+      if (
+        !parsedNonDirective &&
+        this.state.containsOctal &&
+        this.state.octalPosition !== null
```

Comment:
Question: is this just needed for flow, or it's possible that `containsOctal` is `true` but `octalPosition` is false?

If it's just for flow, I have started to use this pattern in other parts of the parser:
```js
/*:: declare var invariant; */ // <-- At the beginning of the file

// ...

if (!parsedNonDirective && this.state.containsOctal) {
  /*:: invariant(this.state.octalPosition !== null) */
  octalPositions.push(this.state.octalPosition);
}
```

Reply:
Thanks for this feedback! I actually think we can get rid of `this.state.octalPosition` altogether. Please let me know what you think!